csharp/src/Ice/Proxy.cs

```diff
@@ -321,31 +321,35 @@ public static class Proxy
         /// <param name="oneway">When true, the request is sent as a oneway request. When false, it is sent as a
         /// two-way request.</param>
         /// <param name="request">The incoming request frame.</param>
+        /// <param name="compressRequestFrame">True if the Ice1 request must use protocol compression, false otherwise.
+        /// </param>
+        /// <param name="compressResponseFrame">True if the Ice1 response must use protocol compression, false
+        /// otherwise.</param>
         /// <param name="progress">Sent progress provider.</param>
         /// <param name="cancel">A cancellation token that receives the cancellation requests.</param>
         /// <returns>A task holding the response frame.</returns>
-        public static ValueTask<OutgoingResponseFrame> ForwardAsync(this IObjectPrx proxy,
-                                                                    bool oneway,
-                                                                    IncomingRequestFrame request,
-                                                                    IProgress<bool>? progress = null,
-                                                                    CancellationToken cancel = default)
+        public static async ValueTask<OutgoingResponseFrame> ForwardAsync(
+            this IObjectPrx proxy,
+            bool oneway,
+            IncomingRequestFrame request,
+            bool compressRequestFrame = false,
+            bool compressResponseFrame = false,
+            IProgress<bool>? progress = null,
+            CancellationToken cancel = default)
         {
             var forwardedRequest = new OutgoingRequestFrame(proxy,
                                                             request.Operation,
                                                             request.IsIdempotent,
+                                                            compressRequestFrame,
```

Comment:
Why didn't we keep the compressionStatus on IncomingRequestFrame and use that to compress or not the forwarded compressed request frame instead of relying on the user to decide on this? For the response, we could also have moved compressionStatus to IncomingFrame and also compress the forwarded response based on this?

Reply:
We decide the client shouldn't mandate the compression, inline with the other change to only use the metadata to decide when the response should be compressed.

Reply:
Because it's not correct: if you receive a compressed ice1 frame, you decompress to read it, and if you forward it again using ice1, there is no reason to automatically recompress it. There is also no benefit: the compressed bytes weren't kept and even if we kept them, they could not be reused.